enumerables.rb

```diff
@@ -0,0 +1,184 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/BlockNesting
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < length
+        if instance_of?(Array)
+          yield self[i]
+        else
+          yield(keys[i], values[i])
+        end
+        i += 1
+      end
+    else
+      puts 'block was not given'
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < length
+        if instance_of?(Array)
+          yield(self[i], i)
+        else
+          yield(values[i], i)
+        end
+        i += 1
+      end
+    else
+      puts 'block was not given'
+    end
+  end
+
+  def my_select
+    if block_given?
+      i = 0
+      new_arr = []
+      new_hash = {}
+      while i < length
+        if instance_of?(Array)
+          new_arr << self[i] if yield self[i]
+        elsif yield(keys[i], values[i])
+          new_hash[keys[i]] = values[i]
+        end
+        i += 1
+      end
+
+      return new_arr if instance_of?(Array)
+      return new_hash if instance_of?(Hash)
+
+    else
+      puts 'block was not given'
+    end
+  end
+
+  def my_all?(sth = nil)
+    return true if empty?
+
+    if block_given?
+      my_each do |item|
+        return false unless yield item
+      end
+    else
+      my_each do |item|
+        if sth.instance_of?(Regexp)
+          return false if item.match?(sth) == false
+        elsif sth.is_a? Class
+          return false unless item.is_a? sth
+        elsif [false, nil].include?(item)
+          return false
+        end
+      end
+    end
+    true
+  end
+
+  def my_any?(sth = nil)
+    return false if empty?
+
+    if block_given?
+      my_each do |item|
+        return true if yield item
+      end
+    else
+      my_each do |item|
+        if sth.instance_of?(Regexp)
+          return true if item.match?(sth)
+        elsif sth.is_a? Class
+          return true if item.is_a? sth
+        else
+          return false if self == [nil] || self == [false] || self == [nil, false] || self == [false, nil]
+
+          return true
+        end
+      end
+    end
+    false
+  end
```

Comment:
- [ ] enumerables #my_any? returns false if the block ever returns a value false or nil::range
      Failure/Error: expect(range.my_any?(&false_block)).to eq(range.any?(&false_block))
      
      NoMethodError:
        undefined method `empty?' for 5..50:Range

- [ ] enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
      Failure/Error: expect(false_array.my_any?).to be false_array.any?
      
        expected false
             got true

- [ ] enumerables #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
      Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
      
        expected false
             got true